### PR TITLE
feat: add an explanation for what each API Key Role is for

### PR DESCRIPTION
**Pull request by dogfrogfog**
#1076 

https://user-images.githubusercontent.com/47758224/177331808-3d7e59c0-85dd-4478-bd06-8dc027e5394c.mov

### CODE REVIEW

File: webapp/javascript/components/Settings/APIKeys/APIKeyAddForm.tsx [hunk 1]
```diff
@@ -101,7 +103,25 @@ function APIKeyAddForm() {
...
+                Role
+                <span
+                  className={styles.infoMark}
+                  onMouseEnter={() => setRolesTooltipVisibility(true)}
+                  onMouseLeave={() => setRolesTooltipVisibility(false)}
+                >
+                  <Tooltip
+                    title={
+                      'Admin - Manage users/API Key using the HTTP API\n' +
+                      'ReadOnly - for visualization only, eg Grafana datasource\n' +
```
Comment (Rperry2174) [inline webapp/javascript/components/Settings/APIKeys/APIKeyAddForm.tsx:116]:
```suggestion
                      'ReadOnly: Used only for visualizations (i.e. Grafana datasource, embedding in a UI)\n' +
```

File: webapp/javascript/components/Settings/APIKeys/APIKeyAddForm.tsx [hunk 2]
```diff
@@ -101,7 +103,25 @@ function APIKeyAddForm() {
...
+              <h4>
+                Role
+                <span
+                  className={styles.infoMark}
+                  onMouseEnter={() => setRolesTooltipVisibility(true)}
+                  onMouseLeave={() => setRolesTooltipVisibility(false)}
+                >
+                  <Tooltip
+                    title={
+                      'Admin - Manage users/API Key using the HTTP API\n' +
```
Comment (Rperry2174) [inline webapp/javascript/components/Settings/APIKeys/APIKeyAddForm.tsx:115]: @eh-am not sure about this but it's just managing users / api keys right? Not just via the http api
```suggestion
                      'Admin: Manage users/API keys\n' +
```

File: webapp/javascript/components/Settings/APIKeys/APIKeyAddForm.tsx [hunk 3]
```diff
@@ -101,7 +103,25 @@ function APIKeyAddForm() {
...
+                      'ReadOnly - for visualization only, eg Grafana datasource\n' +
+                      'Agent - the one you use in your code (https://pyroscope.io/docs/api-key-authentication/)'
```
Comment (Rperry2174) [inline webapp/javascript/components/Settings/APIKeys/APIKeyAddForm.tsx:117]:
```suggestion
                      'Agent: Used for "authentication token" in code when auth is enabled\n'
```

**Review by Rperry2174 — COMMENTED** [review]
Some small changes to text / wording

**Review by kolesnikovae — APPROVED** [review]
Looks nice!

**Comment by Rperry2174** [comment]
Thank @dogfrogfog looks great!

**Pull request merged**